src/components/CodeEditor/CodeEditor.jsx

```diff
@@ -246,7 +264,15 @@ export default function CodeEditor(props) {
 
   useEffect(() => {
     if (editorRef.current) {
-      editorRef.current.setValue(props.defaultCode);
+      const url = window.location.href.split("/").pop();
+      if (sessionStorage.getItem(url) !== props.defaultCode && sessionStorage.getItem(url))
```

Comment:
maybe we don't need to check for the comparison ? if `sessionStorage.getItem(url) === props.defaultCode` then its the same as setting the value to be props.defaultCode. 

```suggestion
      if (sessionStorage.getItem(url))
```